tests/drivers/uart/uart_emul/prj.conf

```diff
@@ -0,0 +1,3 @@
+CONFIG_ZTEST=y
+CONFIG_ZTEST_NEW_API=y
+CONFIG_UART_EMUL=y
```

Comment:
This symbol is selected by default in the device tree, probably you should set `CONFIG_SERIAL=y`?